diff --git a/src/java.base/unix/native/libjava/ProcessImpl_md.c b/src/java.base/unix/native/libjava/ProcessImpl_md.c
--- a/src/java.base/unix/native/libjava/ProcessImpl_md.c
+++ b/src/java.base/unix/native/libjava/ProcessImpl_md.c
@@ -44,7 +44,11 @@
 #include <signal.h>
 #include <string.h>
 #if defined(__solaris__) || defined(_ALLBSD_SOURCE) || defined(_AIX) || defined(__linux__)
+#ifdef __ANDROID__
+#include "posix_spawn.h"
+#else
 #include <spawn.h>
+#endif
 #endif
 
 #include "childproc.h"